src/knx/tpuart_data_link_layer.cpp

```diff
@@ -1,539 +1,546 @@
-    : DataLinkLayer(devObj, addrTab, layer, platform)
-{
-}
-
-void TpUartDataLinkLayer::frameBytesReceived(uint8_t* buffer, uint16_t length)
-{
-    //printHex("=>", buffer, length);
-    CemiFrame frame(buffer, length);
-
-    frameRecieved(frame);
-}
-
-void TpUartDataLinkLayer::dataConBytesReceived(uint8_t* buffer, uint16_t length, bool success)
-{
-    //printHex("=>", buffer, length);
-    CemiFrame frame(buffer, length);
-    dataConReceived(frame, success);
-}
-
-void TpUartDataLinkLayer::enabled(bool value)
-{
-    if (value && !_enabled)
-    {
-        _platform.setupUart();
-
-        if (resetChip()){
-            _enabled = true;
-            print("ownaddr ");
-            println(_deviceObject.induvidualAddress(), HEX);
-        }
-        else{
-        	_enabled = false;
-        	println("ERROR, TPUART not responding");
-        }
-        return;
-    }
-
-    if (!value && _enabled)
-    {
-        _enabled = false;
-        stopChip();
-        _platform.closeUart();
-        return;
-    }
-}
-
-bool TpUartDataLinkLayer::enabled() const
-{
-    return _enabled;
-}
-
-bool TpUartDataLinkLayer::sendSingleFrameByte()
-{
-    uint8_t cmd[2];
-    uint8_t idx = _TxByteCnt / 64;
-
-    if (_sendBuffer == NULL)
-        return false;
-
-    if (_TxByteCnt < _sendBufferLength)
-    {
-        if (idx != _oldIdx)
-        {
-            _oldIdx = idx;
-            cmd[0] = U_L_DATA_OFFSET_REQ | idx;
-            _platform.writeUart(cmd, 1);
-        }
-
-        if (_TxByteCnt != _sendBufferLength - 1)
-            cmd[0] = U_L_DATA_START_CONT_REQ | _TxByteCnt;
-        else
-            cmd[0] = U_L_DATA_END_REQ | _TxByteCnt;
-
-        cmd[1] = _sendBuffer[_TxByteCnt];
-
-        _platform.writeUart(cmd, 2);
-        _TxByteCnt++;
-        return true;
-    }
-    else
-    {
-        _TxByteCnt = 0;
-        return false;
-    }
-}
-
-void TpUartDataLinkLayer::addFrameTxQueue(CemiFrame& frame)
-{
-
-    _tx_queue_frame_t* tx_frame = new _tx_queue_frame_t;
-    tx_frame->length = frame.telegramLengthtTP();
-    tx_frame->data = new uint8_t[tx_frame->length];
-    tx_frame->next = NULL;
-    frame.fillTelegramTP(tx_frame->data);
-
-    if (_tx_queue.back == NULL)
-    {
-        _tx_queue.front = _tx_queue.back = tx_frame;
-    }
-    else
-    {
-        _tx_queue.back->next = tx_frame;
-        _tx_queue.back = tx_frame;
-    }
-}
-
-bool TpUartDataLinkLayer::isTxQueueEmpty()
-{
-    if (_tx_queue.front == NULL)
-    {
-        return true;
-    }
-    return false;
-}
-
-void TpUartDataLinkLayer::loadNextTxFrame()
-{
-    if (_tx_queue.front == NULL)
-    {
-        return;
-    }
-    _tx_queue_frame_t* tx_frame = _tx_queue.front;
-    _sendBuffer = tx_frame->data;
-    _sendBufferLength = tx_frame->length;
-    _tx_queue.front = tx_frame->next;
-
-    if (_tx_queue.front == NULL)
-    {
-        _tx_queue.back = NULL;
-    }
-    delete tx_frame;
-}
-#endif
\ No newline at end of file
+#include "tpuart_data_link_layer.h"
```

Comment:
Something went wrong here. Why is this diff so large? Did you change the line endings? Also why do you want to sent reset every second?

Reply:
I don't know the reason of the large diff, do i need to check that? or you can deal with it?